### ISSUE TITLE
Bug:  Resolve local module in eslint.config.js for flat config will not work

### ISSUE THREAD

**Issue by linlinyang** [eslint/eslint#18127]
### Environment

Node version: 18.17.1
npm version: 9.6.7
Local ESLint version: 8.56.0
Global ESLint version: 8.56.0
Operating System: window 11

### What parser are you using?

@typescript-eslint/parser

### What did you do?

// eslint.config.js
```js
export default [{
  files: ["**/*.?([cm])[jt]s"],
    rules: {
      semi: "error"
  }
}]
```
It will work fine. But:
In config.js file, the same folder
```js
// config.js
export default [{
  files: ["**/*.?([cm])[jt]s"],
    rules: {
      semi: "error"
  }
}]
```
And then import it
```js
// eslint.config.js
import configs from './config'
// import configs from './config.js' failed too.
export default configs
```
**Eslint will work failed.**

// package.json
```json
{
  "name": "eslint-flat-config",
  "version": "1.0.0",
  "private": true,
  "packageManager": "pnpm@8.14.1",
  "description": "The flat config for eslint",
  "type": "module",
}
```

### What did you expect to happen?

JS syntax has error hits without semi.

### What actually happened?

No error hints

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/vitejs-vite-ehy9uw?file=eslint.config.js

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by linlinyang** [comment]
> Hi @linlinyang! There is not enough data to figure out what might be causing the problem. Can you please provide a minimal reproducible example?

[Look this demo](https://stackblitz.com/edit/vitejs-vite-ehy9uw?file=eslint.config.js)

Run `npm run lint` no error hints
And no error hints in vscode

Name: ESLint
ID: dbaeumer.vscode-eslint
Description: Integrates ESLint JavaScript into VS Code.
Version: 2.4.2

**Comment by mdjermanovic** [comment]
Note that in ESLint v8.56.0 and earlier versions, `eslint` command without arguments doesn't lint anything.

I'm getting the expected eslint error messages in your demo when I replace `"lint": "eslint"` with `"lint": "eslint ."`:

```
❯ npm run lint

> eslint-report@0.0.0 lint
> eslint .


/home/projects/vitejs-vite-pejvth/eslint.config.js
  4:40  error  Array.prototype.map() expects a return value from arrow function  array-callback-return

✖ 1 problem (1 error, 0 warnings)
```

https://stackblitz.com/edit/vitejs-vite-pejvth?file=package.json

**Comment by linlinyang** [comment]
@mdjermanovic Thanks.

And how about vscode-eslint does not work

**Comment by mdjermanovic** [comment]
Works for me locally:

![image](https://github.com/eslint/eslint/assets/44349756/b8ed5360-3ee1-4e13-a59c-0ead4d003f7b)

Note that vscode-eslint tracks `eslint.config.js` and automatically reloads eslint if that file has changed, but when you're making changes in another file like `config.js`, you'll probably need to restart VS Code for the new configuration to apply.

**Comment by linlinyang** [comment]
@mdjermanovic Reload VS Code work fine. Thanks!
